### ISSUE TITLE
Find the best faster

### ISSUE THREAD

**Issue by wellington36** [wellington36/extrapolation_methods#8]
In the code, we have the follow problem:

![image](https://user-images.githubusercontent.com/61877847/223845174-1c0533de-dcba-459a-8a85-14968462ac5d.png)

Now we are solving this problem, evaluating one by one, but this complexity is $O(n O(t))$. In this issue we want to propose an algorithm with lower complexity.

**Comment by wellington36** [comment]
The following algorithm solves the above problem with complexity $O(\log^2(n^*))$:

```python
n = n0
i = -1 # trash

while (i not in {0, 1}):
    i = -1 # trash

    while (not t(n)):
        i = i + 1  # i = 0,1,...
        n = n0 + 2^i
    n0 = n0 + 2^(i-1)

return n
```

**Comment by wellington36** [comment]
The following algorithm solves the above problem with complexity $O(2 * \log(n^*))$:

```python
n = n0
i = -1 # trash

while (not t(n)):
    i = i + 1  # i = 0,1,...
    n = n0 + 2^i
n0 = n0 + 2^(i-1)

while (n > n0 ):
    if (t(int((n + n0)/2 + 1)):
        n0 = int((n + n0)/2 + 1)
    else:
        n = int((n + n0)/2 + 1)

return n
```

**Comment by wellington36** [comment]
Closed in ff57d31.
